admin/app/components/solidus_admin/ui/modal/component.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class SolidusAdmin::UI::Modal::Component < SolidusAdmin::BaseComponent
+  renders_one :body
+  renders_one :actions
+
+  def initialize(
+    title:,
+    id: SecureRandom.hex,
+    close_message: "Close",
+    button_component: component("ui/button")
```

Comment:
In line with the recently merged PR #5372, we should now avoid injecting components through the initializer.